app/models/daos/OAuth1InfoDAO.scala

```diff
@@ -0,0 +1,101 @@
+package models.daos
+
+import com.mohiva.play.silhouette.api.LoginInfo
+import com.mohiva.play.silhouette.impl.daos.DelegableAuthInfoDAO
+import com.mohiva.play.silhouette.impl.providers.OAuth1Info
+import javax.inject.Inject
+import play.api.libs.concurrent.Execution.Implicits._
+import play.api.db.slick.DatabaseConfigProvider
+import scala.concurrent.Future
+
+/**
+ * We do not use OAuth, but Silhouette requires implementations for OAuth1InfoDAO and OAuth2InfoDAO.
+ *
+ * This implementation is based on https://github.com/sbrunk/play-silhouette-slick-seed
+ */
+class OAuth1InfoDAO @Inject() (protected val dbConfigProvider: DatabaseConfigProvider)
+    extends DelegableAuthInfoDAO[OAuth1Info] with DAOSlick {
+
+  import driver.api._
+
+  protected def oAuth1InfoQuery(loginInfo: LoginInfo) = for {
+    dbLoginInfo <- loginInfoQuery(loginInfo)
+    dbOAuth1Info <- slickOAuth1Infos if dbOAuth1Info.loginInfoId === dbLoginInfo.id
+  } yield dbOAuth1Info
+
+  // Use subquery workaround instead of join to get authinfo because slick only supports selecting
+  // from a single table for update/delete queries (https://github.com/slick/slick/issues/684).
+  protected def oAuth1InfoSubQuery(loginInfo: LoginInfo) =
+    slickOAuth1Infos.filter(_.loginInfoId in loginInfoQuery(loginInfo).map(_.id))
+
+  protected def addAction(loginInfo: LoginInfo, authInfo: OAuth1Info) =
+    loginInfoQuery(loginInfo).result.head.flatMap { dbLoginInfo =>
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Usage of head on collections.](https://www.codacy.com/app/HiP-App/HiPCMS/file/2283741786/issues/source?bid=3144114&fileBranchId=3144125#l32)